tests/basic.rs

```diff
@@ -85,6 +85,14 @@ fn write_file_with() -> Result<()> {
     })?;
     let actual_contents = std::fs::read_to_string(td.path().join(testname))?;
     assert_eq!(testcontents2, actual_contents.as_str());
+    let srcf = d.open_file("testfile")?;
+    d.write_file_with("testfile2", 0o644, |w| -> anyhow::Result<()> {
+        let mut bufr = std::io::BufReader::new(srcf);
```

Comment:
@lucab here we're moving the `File` instance into the closure, it compiles now with `FnOnce` and I also reverse-verified we fail to compile this if it's just `Fn`.

Reply:
Ack, thanks! I think we'd need a semver-bump for this though, right? (Which is why I went for a quicker workaround in rpm-ostree initially).

Additionally, I initially had a `FnMut` in mind here in order to get some results too (e.g. `written_entries_accum`). Is there some drawback in going that route?

Reply:
I think this is semver compatible because: https://doc.rust-lang.org/std/ops/trait.FnOnce.html says:

> Since both Fn and FnMut are subtraits of FnOnce, any instance of Fn or FnMut can be used where a FnOnce is expected.

And this also relates to `FnMut` vs `FnOnce` - since we do only call this once, that's the most permissive on our side and hence most flexible for the caller, so we should prefer that.